### ISSUE TITLE
Shader compilation: reading 0 as int instead of float

### ISSUE THREAD

**Issue by Gerwin2k** [luciusDXL/TheForceEngine#334]
Thanks to the new shader error logging I could trace this.
```
error(#162) Wrong operand types:
no operation ":" exists that takes a left-hand operand of type "highp float" and a right operand of type "const int" 
```
Only an issue with older Windows Radeon Drivers for cards like HD 6xxx / HD 7xxx. 
Seemingly fixed by making the 0 into a 0.0. Which makes the shader compiler recognize it as float.
Fixes true color rendering for my system. Also seems to speed up 8-bit GPU rendering, so I figure the system was somehow tripping over it constantly?

Fix in Shaders\textureSampleFunc.h

```
// GB 2023, changed 0 to 0.0 four times:
vec2 wrapCoordF(vec2 uv, vec2 edge)
{
	uv += vec2(0.5,-0.5);

	uv = fmod(uv, edge);
	uv.x += (uv.x < 0.0) ? edge.x : 0.0;
	uv.y += (uv.y < 0.0) ? edge.y : 0.0;

	uv -= vec2(0.5,-0.5);
	return uv;
}

// GB 2023, changed 0 to 0.0 four times:
// Approximate the distortion that happens in software when floor and ceiling textures are not 64x64.
vec2 wrapCoordFlatF(vec2 uv, vec2 edge)
{
	float coord = fmod(uv.x, 64.0) * 64.0 + fmod(uv.y, 64.0);
	uv = wrapCoordF(vec2(coord / edge.y, coord), edge);
	uv.x += (uv.x < 0.0) ? edge.x : 0.0;
	uv.y += (uv.y < 0.0) ? edge.y : 0.0;
	return uv;
}
```

**Comment by luciusDXL** [comment]
*sigh* - GLSL compilers strike again. Ok, I will post a fix tonight (right now that file has other changes that I have to finish first).

**Comment by Gerwin2k** [comment]
No hurry.

The retro WIN32 / XP build, that worked on, has this change already.

Just thought it would be useful to post the finding here, for whatever it is worth.

**Comment by luciusDXL** [comment]
Fixed in the 1.09.52 release.
